socorro/processor/mozilla_transform_rules.py

```diff
@@ -606,75 +608,68 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        self.version_string_api = config.version_string_api
+        self.cache = ExpiringCache(max_size=self.CACHE_MAX_SIZE, ttl=self.CACHE_TTL)
 
     def version(self):
         return '1.0'
 
     def _get_version_data(self, product, version, build_id):
-        """Return the real version number of a specific product, version and
-        build.
+        """Return the real version number of a specific product, version and build
+
+        For example, beta builds of Firefox declare their version number as the
+        major version (i.e. version 54.0b3 would say its version is 54.0). This
+        database call returns the actual version number of said build (i.e.
+        54.0b3 for the previous example).
+
+        :arg product: the product
+        :arg version: the version as a string. e.g. "56.0"
+        :arg build_id: the build_id as a string.
+
+        :returns: ``None`` or the version string that should be used
+
+        :raises requests.RequestException: raised if it has connection issues with
+            the host specified in ``version_string_api``
 
-        For example, beta builds of Firefox declare their version
-        number as the major version (i.e. version 54.0b3 would say its version
-        is 54.0). This database call returns the actual version number of said
-        build (i.e. 54.0b3 for the previous example).
         """
+        if not (product and version and build_id):
+            return None
+
         key = '%s:%s:%s' % (product, version, build_id)
+        if key in self.cache:
+            return self.cache[key]
 
-        if key in self._versions_data_cache:
-            return self._versions_data_cache[key]
-
-        sql = """
-            SELECT
-                pv.version_string
-            FROM product_versions pv
-                LEFT JOIN product_version_builds pvb ON
-                    (pv.product_version_id = pvb.product_version_id)
-            WHERE pv.product_name = %(product)s
-            AND pv.release_version = %(version)s
-            AND pvb.build_id = %(build_id)s
-        """
-        params = {
+        # FIXME(willkg): take the request/retry code in socorro/scripts/ and
+        # put that in lib. Then reuse it here.
+        session = session_with_retries(self.version_string_api)
+
+        resp = session.get(self.version_string_api, params={
             'product': product,
             'version': version,
-            'build_id': build_id,
-        }
-        results = self.transaction(
-            execute_query_fetchall,
-            sql,
-            params
-        )
-        for real_version, in results:
-            self._versions_data_cache[key] = real_version
+            'build_id': build_id
+        })
 
-        return self._versions_data_cache.get(key)
+        if resp.status_code == 200:
+            hits = resp.json()['hits']
+            if hits:
+                self.cache[key] = hits[0]
+                return hits[0]
```

Comment:
I need to fix this. If we get back an HTTP 200 and there are *no hits*, we should cache that response, too.